tests/grid/flatten_grid_01.output

```diff
@@ -8,8 +8,8 @@ $NOD
 $ENDNOD
 $ELM
 2
-1 1 0 0 2 1 3 
-2 1 0 0 2 3 2 
+1 1 0 0 2 1 3
+2 1 0 0 2 3 2
```

Comment:
This, and likely corresponding changes in other tests, is not correct. The test outputs in gmsh .msh format, not in UCD format. Its output should not have changed. This is because in the `GridOut::write_msh()` function, we make reference to the `ucd_to_deal` variable, but I think that needs to become a local variable with the old ordering.

Reply:
The only change here is the trailing whitespace. I use a script to remove it whenever I add changes, in this case it leads to changes in unexpected places. I'll rework this and see wether these unintended changes persist.

Reply:
But there are many other places where there are actual changes.

I think that the general comment about the MSH output using the UCD array remains true.

Reply:
Right, I see what you mean. The gmsh node numbering is compatible with the correct UCD numbering though, as you can see [here](https://gmsh.info/doc/texinfo/gmsh.html#Node-ordering). For example checking `tests/grid/grids/grid_in_msh_01.3d.msh` you can observe that the volume is negative too. So I think the changes are not really unexpected?
![20230126_11h47m24s_grim](https://user-images.githubusercontent.com/20728434/214817560-7edfe125-af8b-440d-a9ec-2bcf1952b144.png)

Reply:
So do I understand you correctly that the gmsh output was *also* wrong before?

Reply:
@nlsmrg Separately, two things. 

First, many thanks for the investigating! Your help here is greatly appreciated!

Second, looking at this again, I am very uncomfortable merging this big a patch. Could I suggest the following approach:
* Introduce a local translation array in `GridOut` for the moment that keeps the msh output as it was before, so that the changes in this patch really only affect the UCD format. This would make it much easier to convince ourselves that there are no unintended side effects.
* In a second pull request, we remove the local translation array again for the msh format. This should allow us to verify that that change really only affects the msh output and nothing else.

Reply:
> So do I understand you correctly that the gmsh output was _also_ wrong before?

Yes I'm certain this is the case. I do agree that fixing this in a single patch is risky, so lets go with your suggestion.

Reply:
@nlsmrg  Have you had time to think about this?

Reply:
I made some progress, but I have a lot of other stuff to do right now. I think the last thing missing is related to reading the Abaqus format. Maybe thats not as straightforward as the other cases due to this [Abaqus_to_UCD](https://github.com/dealii/dealii/blob/1d44a69b70227e93bc85e7ac0d6aba632a149438/source/grid/grid_in.cc#L1133)  conversion? I didn't look into it yet.